dealer/dealer.py

```diff
@@ -31,31 +33,76 @@ def make_move(
             # If the action can change our count
             if action.action_taken in [Item.Beer, Shoot.You, Shoot.Opponent]:
                 self.known_shell = None
-                if action_outcome == Shell.Live:
-                    self.live_count -= 1
-                elif action_outcome == Shell.Blank:
-                    self.blank_count -= 1
+                if self.used_inverter:
+                    if action_outcome == Shell.Live:
+                        self.blank_count -= 1
+                    if action_outcome == Shell.Blank:
+                        self.live_count -= 1
+                else:
+                    if action_outcome == Shell.Live:
+                        self.live_count -= 1
+                    elif action_outcome == Shell.Blank:
+                        self.blank_count -= 1
+                self.used_inverter = False
+            if action.action_taken == Item.Inverter:
+                self.used_inverter = not self.used_inverter
+
             if action.action_taken == Item.Magnifier:
                 if action_outcome != Shell.Unknown:
                     self.known_shell = action_outcome
+            if action.action_taken == Item.Phone:
+                if isinstance(action_outcome, PhoneCall):
+                    # self.memory_phone_call = action_outcome
+                    if action_outcome.number == 0:
+                        self.known_shell = action_outcome.shell
+
         if not available:
             return Nothing()
         guaranteed_lethal = self.blank_count == 0 or self.known_shell == Shell.Live
         if Item.Cigarettes in available and my_hp < 4:
             return Item.Cigarettes
+        if Item.Medicine in available and my_hp < 4:
```

Comment:
I played the original game a bit and never came across the dealer using medicine on 1 HP. Maybe it is a restriction for him?
I suggest something like
`if Item.Medicine in available and my_hp < 4 and my_hp > 1:`
Correct me if I am wrong

Reply:
It might be true, but cannot fully test it out.
It may not be a priority for the dealer to use the medicine on 1HP but maybe it could use the medicine by random.
```suggestion
        if Item.Medicine in available and 1 < my_hp < 4:
```
I will keep the following unchanged for now.
https://github.com/nokrolikno/Buckshot-roulette/blob/7d5d2cd2ce0f0fc207e0b96889f7825c6ff3ea68/dealer/dealer.py#L121-L122
Maybe more precise dealer logic can be done in future commits and PRs.